api/utils.go

```diff
@@ -240,14 +240,17 @@ func hijack(tlsConfig *tls.Config, addr string, w http.ResponseWriter, r *http.R
 	if err != nil {
 		return err
 	}
+
 	hj, ok := w.(http.Hijacker)
 	if !ok {
-		return err
+		return errors.New("Failed to assert into http.Hijacker")
```

Comment:
This message should be something like `Docker server does not support hijacking`.

The assertion is checking that we can hijack the connection, because we will try to hijack it.

docker-engine does
```
	conn, _, err := w.(http.Hijacker).Hijack()
	if err != nil {
		return nil, nil, err
	}
```
[here](https://github.com/docker/docker/blob/master/api/server/httputils/httputils.go#L29)